### PR TITLE
Update for Middleman v4

### PR DESCRIPTION
**Pull request by angeloashmore**
Compatible with Middleman v4. Utilizes Cucumber for tests.

### CODE REVIEW

**Comment by Aupajo** [comment]
Hi @angeloashmore! Thanks for your pull request.

I'd like to merge this, but I'd also like to stick with plain RSpec.

I appreciate that many extensions in the Middleman community use Cucumber but I would not like to be the maintainer of one that does.

Briefly, my objection is that Cucumber is an acceptance test framework designed specifically for involving both developers and non-technical stakeholders – it provides a common language they can use to agree on when a particular use case is “done”. Using it to drive integration tests seems like an abuse of that design (we're all developers here), and an unnecessary one at that. It's slower, requires additional abstraction, and requires learning another framework, all without a clear benefit (if it's clarity your after, you should be striving to do that with code). Beyond this, while integration tests are important, they should never be the whole story: both unit tests and integration tests are needed.

I understand if this might come across as too academic or pedantic, and I don't want to hold up new changes, so I'll make this promise: I'll endeavour to check out this PR and backport it to RSpec. If I don't get there within the next 24 hours, I'll merge it as-is.

Thanks, and thanks for understanding!

**Comment by angeloashmore** [comment]
No problem, I understand. I've never used Cucumber myself either, but it appeared to be the standard for Middleman extensions. The existing tests weren't running (I'm assuming the API for Middleman changed) so I copied the test structure from another Middleman extension.

File: lib/middleman-search_engine_sitemap/extension.rb [hunk 1]
```diff
@@ -49,14 +41,7 @@ def not_excluded?(resource)
...
 
       def page_ext
-        File.extname(app.index_file)
-      end
-
-      def register_extension_templates
-        # We call reload_path to register the templates directory with Middleman.
-        # The path given to app.files must be relative to the Middleman site's root.
-        templates_dir_relative_from_root = Pathname(TEMPLATES_DIR).relative_path_from(Pathname(app.root))
-        app.files.reload_path(templates_dir_relative_from_root)
```
Comment (Aupajo) [inline lib/middleman-search_engine_sitemap/extension.rb]: Is this no longer needed in 4.0?

Comment (angeloashmore) [inline lib/middleman-search_engine_sitemap/extension.rb]: It seems to work fine without it. The Middleman API no longer has `app.files.reload_path` as far as I can tell.

Comment (Aupajo) [inline lib/middleman-search_engine_sitemap/extension.rb]: This was my workaround for https://github.com/middleman/middleman/issues/1217. It's a hard thing to test for, and there is no test for it at the moment. The problem arises when this gem gets placed in the `vendor` directory of a Middleman site (platforms like Heroku do this).

**Comment by Aupajo** [comment]
I've done some experiments and not been able to replicate the Heroku issue (#2, #5). It also looks as though FileWatcher is now [no longer ignoring `vendor`](https://github.com/middleman/middleman/blob/6d1c3562a705d06b4f8e09276dc12a10c80c76b4/middleman-core/lib/middleman-core/core_extensions/file_watcher.rb#L19-L24), which bodes well. Going to do some Heroku testing next.

**Comment by Aupajo** [comment]
Looks good! I've tested it on Heroku under the slightly old instructions from [this blog](https://blog.pixelcab.in/2015/03/23/blogging-on-heroku-with-middleman/) and in the new Rack mode, and both work fine under `vendor`.

@angeloashmore, I'm close to shipping this. If you have a spare moment, I'd love a second confirmation that it works for someone else :)

```ruby
gem 'middleman-search_engine_sitemap', '~> 1.4.0.beta'
```

**Comment by angeloashmore** [comment]
Thanks for taking the time to work on this. I just tested it with padrino/padrino-web and everything looks good!

**Comment by Aupajo** [comment]
Thanks for your help, @angeloashmore! Version `1.4.0` now supports Middleman 4.

```ruby
gem 'middleman-search_engine_sitemap', '~> 1.4.0'
```

**Pull request closed without merging**